src/elements/content-sidebar/activity-feed/annotations/AnnotationActivity.js

```diff
@@ -16,44 +17,73 @@ import { ACTIVITY_TARGETS } from '../../../common/interactionTargets';
 import { PLACEHOLDER_USER } from '../../../../constants';
 import type { Annotation, AnnotationPermission } from '../../../../common/types/feed';
 import type { GetAvatarUrlCallback, GetProfileUrlCallback } from '../../../common/flowTypes';
-import type { User } from '../../../../common/types/core';
+import type { SelectorItems, User } from '../../../../common/types/core';
 
 import './AnnotationActivity.scss';
 
 type Props = {
     currentUser?: User,
     getAvatarUrl: GetAvatarUrlCallback,
+    getMentionWithQuery?: Function,
     getUserProfileUrl?: GetProfileUrlCallback,
     isCurrentVersion: boolean,
+    isDisabled?: boolean,
     item: Annotation,
+    mentionSelectorContacts?: SelectorItems<>,
     onDelete?: ({ id: string, permissions: AnnotationPermission }) => any,
+    onEdit?: (id: string, text: string, hasMention: boolean, permissions: AnnotationPermission) => any,
     onSelect?: (annotation: Annotation) => any,
 };
 
 const AnnotationActivity = ({
+    currentUser,
+    isDisabled,
     item,
     getAvatarUrl,
+    getMentionWithQuery,
     getUserProfileUrl,
     isCurrentVersion,
+    mentionSelectorContacts,
     onDelete = noop,
+    onEdit = noop,
     onSelect = noop,
 }: Props) => {
+    const [isInputOpen, setIsInputOpen] = React.useState(false);
+    const [isEditing, setIsEditing] = React.useState(false);
     const { created_at, created_by, description, error, file_version, id, isPending, permissions = {}, target } = item;
+    const message = (description && description.message) || '';
 
     const handleDeleteConfirm = (): void => {
         onDelete({ id, permissions });
     };
 
+    const handleEditClick = () => {
+        setIsEditing(true);
+    };
+
     const handleOnSelect = () => {
         onSelect(item);
     };
 
+    const commentFormFocusHandler = (): void => setIsInputOpen(true);
+
+    const commentFormCancelHandler = (): void => {
+        setIsInputOpen(false);
+        setIsEditing(false);
+    };
+
+    const commentFormSubmitHandler = ({ text, hasMention }): void => {
+        setIsInputOpen(false);
+        setIsEditing(false);
+        onEdit(id, text, hasMention, permissions);
+    };
+
     const createdAtTimestamp = new Date(created_at).getTime();
     const createdByUser = created_by || PLACEHOLDER_USER;
     const canDelete = permissions.can_delete;
+    const canEdit = permissions.can_edit;
```

Comment:
Maybe these could be destructured together? Like `const { can_delete: canDelete, can_edit: canEdit } = permissions`?